### ISSUE TITLE
using the ASRN scheme only

### ISSUE THREAD

**Issue by munziliashali** [Canjie-Luo/MORAN_v2#31]
Hi @Canjie-Luo,
thank you for this repo and paper. I am working on text recognition on my custom dataset that consist of letter-only words and letter-and-number words, with length of words of maximum 50 characters. So far using meijieru's repo, the result doesn't seem very ok, as the accuracy just reach 40%.

So, I apply MORAN, but got this error:
![screen shot 2019-02-08 at 2 01 50 pm](https://user-images.githubusercontent.com/40015034/52463624-2bc0c880-2baa-11e9-8818-f211ee3b32bb.png)

do you have an idea how to fix it?
another question, do you think it is feasible to do ASRN only? 

Thank you.

**Comment by Canjie-Luo** [comment]
Please make sure your code is up-to-date and PyTorch version is 0.3. The best way to check out bugs is running our demo.

It's possible to use ASRN only. Just drop MORN in `moran.py`.

**Comment by munziliashali** [comment]
Yes, I used Pytorch ver 0.4, so I must change some code from Tensor to tuple of integer. 

One more question though, can I ask why did you use CrossEntropyLoss, instead of CTCLoss like meijieru's? I couldn't catch the reason in the paper..

Thank you.

**Comment by Canjie-Luo** [comment]
We use a decoder based on the attention mechanism. There is a one-to-one correspondence between the predictions and labels in attention mechanism, while in CTC Loss, the length of predictions is not the same with that of the corresponding labels. That's why we use the Cross Entropy Loss.

You're welcome!
